dronegen/misc.go

```diff
@@ -20,9 +20,74 @@ import (
 	"strings"
 )
 
-func promoteBuildPipeline() pipeline {
+func promoteBuildPipelines() []pipeline {
 	aptPipeline := promoteAptPipeline()
-	return aptPipeline
+	dockerPipeline := promoteDockerPipeline()
+	return []pipeline{aptPipeline, dockerPipeline}
+}
+
+func promoteDockerPipeline() pipeline {
+	dockerPipeline := newKubePipeline("promote-docker")
+	dockerPipeline.Trigger = triggerPromote
+	dockerPipeline.Trigger.Target.Include = append(dockerPipeline.Trigger.Target.Include, "promote-docker")
+	dockerPipeline.Workspace = workspace{Path: "/go"}
+
+	// Add docker service
+	dockerPipeline.Services = []service{
+		dockerService(),
+	}
+	dockerPipeline.Volumes = dockerVolumes()
+
+	dockerPipeline.Steps = append(dockerPipeline.Steps, verifyTaggedBuildStep())
+	dockerPipeline.Steps = append(dockerPipeline.Steps, waitForDockerStep())
+
+	// Pull/Push Steps
+	dockerPipeline.Steps = append(dockerPipeline.Steps, step{
+		Name:  "Pull/retag Docker images",
+		Image: "docker",
+		Environment: map[string]value{
+			"AWS_ACCESS_KEY_ID":          {fromSecret: "TELEPORT_BUILD_USER_KEY"},
+			"AWS_SECRET_ACCESS_KEY":      {fromSecret: "TELEPORT_BUILD_USER_SECRET"},
+			"DOCKER_PRODUCTION_USERNAME": {fromSecret: "PRODUCTION_QUAYIO_DOCKER_USERNAME"},
+			"DOCKER_PRODUCTION_PASSWORD": {fromSecret: "PRODUCTION_QUAYIO_DOCKER_PASSWORD"},
+		},
+		Volumes: dockerVolumeRefs(),
+		Commands: []string{
+			"apk add --no-cache aws-cli",
+			"export VERSION=${DRONE_TAG##v}",
+			// authenticate with staging credentials
+			"aws ecr get-login-password --region=us-west-2 | docker login -u=\"AWS\" --password-stdin 146628656107.dkr.ecr.us-west-2.amazonaws.com",
+			// pull 'temporary' CI-built images
```

Comment:
```suggestion
			// pull staging images
```
I don't think they're temporary -- they should have roughly the same lifecycle as the promoted images.